### ISSUE TITLE
Structs as generic arguments are treated like classes

### ISSUE THREAD

**Issue by VladimirAngelov** [bridgedotnet/Bridge#1180]
If you use a struct as a generic argument, it will be passed by reference everywhere and the $clone method will not be inserted anywhere.

### Expected

0
5

### Actual

5
5

### Steps To Reproduce

[Live](http://live.bridge.net/#f01f1e291e860821a6ab1b0913c888c7)

```csharp
public struct Vector2
{
    public float x;
    public float y;
}

public class App
{
    [Ready]
    public static void Main()
    {
        List<Vector2> list = new List<Vector2>();
        list.Add(new Vector2() { x = 0.0f, y = 1.0f });
        
        Vector2 vec = list[0];
        
        vec.x = 5.0f;
        
        Window.Alert(list[0].x.ToString()); // Oops.
        Window.Alert(vec.x.ToString());
    }
}
```

**Comment by DaniilVeriga** [comment]
Thank you for the report! We'll try to incorporate a fix into 1.12.1 release.

Added a Live test case to the original description.

**Comment by geoffreymcgill** [comment]
Looks like this has been fixed. Our plan is to include within the upcoming 1.12.1 release (<48 hours).
